examples/llama/main.zig

```diff
@@ -1,156 +1,228 @@
 const asynk = @import("async");
-const flags = @import("tigerbeetle/flags");
+const clap = @import("clap");
 const std = @import("std");
 const stdx = @import("stdx");
 const zml = @import("zml");
 
-const llama_mod = @import("llama.zig");
+const llama = @import("llama.zig");
 
-const LlamaLM = llama_mod.LlamaLM;
-const Llama = llama_mod.Llama;
-const KvCache = llama_mod.KvCache;
-const TransformerLayer = llama_mod.TransformerLayer;
-const SelfAttn = llama_mod.SelfAttn;
+const LlamaLM = llama.LlamaLM;
+const Llama = llama.Llama;
+const KvCache = llama.KvCache;
+const TransformerLayer = llama.TransformerLayer;
+const SelfAttn = llama.SelfAttn;
 const Buffer = zml.Buffer;
 const Tensor = zml.Tensor;
 const ShapeOf = zml.ShapeOf;
 
 const log = std.log.scoped(.llama);
 
-const eos_tokens: [3]i32 = .{ 128001, 128008, 128009 };
-
 // set this to false to disable the verbose logging
-const show_mlir = true;
+const show_mlir = false;
 
 pub const std_options = .{
     .log_level = .warn,
     .log_scope_levels = &[_]std.log.ScopeLevel{
-        .{ .scope = .zml_module, .level = if (show_mlir) .debug else .warn },
+        .{ .scope = .@"zml/module", .level = if (show_mlir) .debug else .warn },
         .{ .scope = .llama, .level = .info },
     },
-    .logFn = asynk.logFn,
 };
 
+pub fn tokenize(allocator: std.mem.Allocator, tokenizer: zml.tokenizer.Tokenizer, config: LlamaLM.Config, prompt: []const u8) ![]u32 {
```

Comment:
maybe name tokenizerPrompt or smth cause this is doing more than just tokenizing. Also this feels very llama3-instruct specific. Are we fine with that ?

Reply:
i'll rename it. we only have llama 3 Instruct models in the build.bazel, so it's not sooooo bad it's called unconditionally.